src/game/assets.rs

```diff
@@ -0,0 +1,69 @@
+use bevy::prelude::*;
+
+#[derive(Resource, Reflect)]
+pub struct ImageAssets {
+    pub ducky: Handle<Image>,
+}
+
+impl ImageAssets {
+    pub fn new(asset_server: &AssetServer) -> Self {
+        Self {
+            ducky: asset_server.load("images/ducky.png"),
+        }
+    }
+
+    pub fn all_loaded(&self, assets: &Assets<Image>) -> bool {
+        assets.contains(&self.ducky)
+    }
+}
+
+#[derive(Resource, Reflect)]
+pub struct SfxAssets {
+    pub button_hover: Handle<AudioSource>,
+    pub button_press: Handle<AudioSource>,
+    pub step1: Handle<AudioSource>,
```

Comment:
These make more sense as a HashSet of handles.

Reply:
You mean a `HashMap`, I suppose?